docs/sourcemaps.asciidoc

```diff
@@ -1,38 +1,201 @@
-While improving performance, minified code can be hard to debug.
-A source map library helps by mapping the minified files back to the the original source code.
+++++
+<titleabbrev>Create and upload source maps (RUM)</titleabbrev>
+++++
 
-APM Server provides a <<sourcemap-api,Source Map API>> for uploading source maps.
+Minifying JavaScript bundles is a common practice;
+it can greatly improve the load time and network latency of your applications.
+The problem with minifying code is that it can be hard to debug.
+Here's an example of an exception stack trace in the APM app when using minified code:
 
-By default, <<config-sourcemapping-enabled>> is set to `true`.
-This means if a source map has previously been uploaded,
-source mapping will automatically be applied all incoming transactions and errors.
+[role="screenshot"]
+image::images/source-map-before.png[APM app without source mapping]
 
+It's not very helpful.
+That’s why Elastic APM supports source maps.
+A source map maps minified files back to the original source code, allowing you to maintain the speed advantage of minified code,
+without losing the ability to quickly and easily debug your application.
+Here's the same example as before, but with a source map uploaded and applied:
+
+[role="screenshot"]
+image::images/source-map-after.png[APM app with source mapping]
+
+Follow the steps below to enable source mapping your error stack traces in the APM app:
+
+* <<sourcemap-rum-initialize>>
+* <<sourcemap-rum-generate>>
+* <<sourcemap-rum-upload>>
+
+[float]
+[[sourcemap-rum-initialize]]
+=== Initialize the RUM Agent
+
+First, set the service version of your application while initializing the RUM Agent.
+
+[source,js]
+----
+import { init as initApm } from '@elastic/apm-rum'
+
+const apm = initApm({
+  serviceName: 'myService',
+  serviceVersion: serviceVersion
+})
+----
+
+This `serviceVersion` could be the version from your `package.json`. For example:
+
+[source,js]
+----
+const serviceVersion = require("./package.json").version
+----
+
+Or, it could be a git commit reference. For example:
+
+[source,js]
+----
+const git = require('git-rev-sync')
+const serviceVersion = git.short()
+----
+
+It can also be any other unique string that indicates a specific version of your application.
+APM Server uses this `serviceVersion` to match the correct source map file to each stack trace.
+
+[float]
+[role="child_attributes"]
+[[sourcemap-rum-generate]]
+=== Generate a source map
+
+Source maps can be generated and configured in many different ways.
+To be compatible with Elastic APM, they must follow the
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[source map revision 3 proposal spec].
+
+.Click here to expand a sample source map file
+[%collapsible%closed]
+=====
+[source,json]
+----
+include::./data/intake-api/generated/sourcemap/bundle.js.map[]
+----
+=====
```

Comment:
I'm not sure if we need to reference an example sourcemap, but it's also closed by default so 🤷‍♂️

Reply:
Hah yeah, that was the same thought process I went through, and why I made it collapsed by default. I'll remove it from here. The link at the bottom of the guide to the [source map indices page](https://www.elastic.co/guide/en/apm/server/7.7/sourcemap-indices.html#sourcemap-example) is probably enough.